index.js

```diff
@@ -42,13 +52,44 @@ exports.register = (server, options, next) => {
                 if(!route.config.validate) {
                     route.config.validate = {};
                 }
-                
+
                 route.config.validate.headers = Joi.object({
                         Authorization: Joi.string().description('JWT token')
                     }).unknown();
             }
 
             server.route(route);
+            if(route.handler) {
+                if(getParamNames(route.handler).length < 2) {
+                    // default signature is (request, reply)
+                    // if reply is omitted, then the return value should be interpreted
+                    // as the reply.
+                    // we wrap the handler function to support this
+                    let orignalHandler = route.handler;
+                    route.handler = (request, reply) => {
+                        let retValue = orignalHandler.call(request, reply);
+                        if(retValue instanceof Promise) {
+                          return retValue
+                          .then(reply)
+                          .catch(err => {
+                              // handle promise rejection
+                              if(err && err.isBoom) {
+                                  // boom errors are put through
+                                  return reply(err);
```

Comment:
I'd leave logging Boom errors to the handlers, so it'd be consistent with the conventional way (using reply callback) - the Boom errors aren't autamatically logged.
But it might come in handy to have the server log all errors for sure.